docs/ref/contrib/gis/gdal.txt

```diff
@@ -611,6 +611,26 @@ coordinate transformation:
         >>> polygon.geom_count
         1
 
+    .. attribute:: has_curve
+
+    .. versionadded:: 5.1
+
+    A boolean indicating if this geometry is or has curve geometry.
+
+    .. method:: get_linear_geometry
+
+    .. versionadded:: 5.1
+
+    Returns a linear version of the geometry. If no conversion can be made the
+    original geometry is returned.
```

Comment:
I'm a little worried about this statement, could we, in any way, signal to the caller whether the conversion could be made? Could return a boolean with the result?

Reply:
Looking at this again, I think curved -> linear is likely always going to work, but linear -> curved may not (for example if it's just a straight line!). 

Maybe I should have stopped after the first sentence or could add something like this (from GDAL docs)

> Returns a geometry that has no CIRCULARSTRING, COMPOUNDCURVE, CURVEPOLYGON, MULTICURVE or MULTISURFACE in it, by approximating curve geometries.

But we should consider the linear -> curve case.

Currently the API here is similar to the GDAL API, albeit with 'pythonic' names. As far as I'm aware we've generally stayed close to mirroring the GDAL API. Checking if two geometries are the same is also very straight forward if users wanted to do that. I'm therefore rather unsure about this. 

@claudep would welcome your opinion on this.

Reply:
Not specifically. Maybe I can add some more tests. 

``` python
>>> geom = OGRGeometry("LINESTRING (0 0, 1 0, 2 0)")
>>> geom2 = geom.get_curve_geometry()
>>> geom2.wkt
'LINESTRING (0 0,1 0,2 0)'
```